CompilerSource/compiler/components/write_object_data.cpp

```diff
@@ -720,10 +720,11 @@ static inline void write_script_implementations(ofstream& wto, EnigmaStruct *es,
       wto << comma << "variant argument" << argn;
       comma = ", ";
     }
-    wto << ")\n{\n  ";
+    wto << ")\n{\n";
     if (mode == emode_debug) {
-      wto << "enigma::debug_scope $current_scope(\"script '" << es->scripts[i].name << "'\");\n";
+      wto << "  enigma::debug_scope $current_scope(\"script '" << es->scripts[i].name << "'\");\n";
     }
+    wto << "  ";
```

Comment:
This generally means the logic is not self-contained and should be refactored, but for now, I suppose this is fine. What normally ends up happening when you do things like this is that one line will be indented an extra level, because it's inheriting the indent that the previous block printed for it for some reason.

It looks like this problem runs pretty deep, though, and I don't think it's worth any more time to correct it right now.